core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -80,33 +79,55 @@
 
     private static final Logger logger = LoggerFactory.getLogger(Optimizely.class);
 
-    @VisibleForTesting final DecisionService decisionService;
+    @VisibleForTesting DecisionService decisionService;
     @VisibleForTesting final EventFactory eventFactory;
-    @VisibleForTesting final ProjectConfig projectConfig;
+    @VisibleForTesting ProjectConfig projectConfig;
     @VisibleForTesting final EventHandler eventHandler;
     @VisibleForTesting final ErrorHandler errorHandler;
+    public boolean isValid = false;
     public final NotificationCenter notificationCenter = new NotificationCenter();
 
     @Nullable private final UserProfileService userProfileService;
 
-    private Optimizely(@Nonnull ProjectConfig projectConfig,
-                       @Nonnull DecisionService decisionService,
-                       @Nonnull EventHandler eventHandler,
+    private Optimizely(@Nonnull EventHandler eventHandler,
                        @Nonnull EventFactory eventFactory,
                        @Nonnull ErrorHandler errorHandler,
+                       @Nullable DecisionService decisionService,
                        @Nullable UserProfileService userProfileService) {
-        this.projectConfig = projectConfig;
         this.decisionService = decisionService;
         this.eventHandler = eventHandler;
         this.eventFactory = eventFactory;
         this.errorHandler = errorHandler;
         this.userProfileService = userProfileService;
     }
 
-    // Do work here that should be done once per Optimizely lifecycle
+    /**
+     * Initializes the SDK state. Can conceivably re-use this in the future with datafile sync where
+     * we can re-initialize the SDK instead of re-instantiating.
+     */
     @VisibleForTesting
-    void initialize() {
+    void initialize(@Nonnull String datafile, @Nullable ProjectConfig projectConfig) {
+        if (projectConfig == null) {
```

Comment:
This step de-couples the "building" of the Optimizely instance from the "initialization", which is where we start populating state.